lib/engine/train.rb

```diff
@@ -30,6 +30,7 @@ def initialize(name:, distance:, price:, index: 0, **opts)
       @operated = false
       @events = (opts[:events] || []).select { |e| @index == (e['when'] || 1) - 1 }
       @reserved = opts[:reserved] || false
+      @requires_token = opts[:requires_token].nil? ? true : opts[:requires_token]
```

Comment:
I'm not a Ruby expert, but couldn't you just add "requires_token: true" to the def initialize for the train class?

Reply:
Otherwise, looks OK to me.

Reply:
> requires_token: true

Sure can. I was following the pattern of the reserved variable, but that is cleaner.

Reply:
@tobymao Do you have a cleaner way to write this line that I removed?

Reply:
!!opts[:requireds_token]

Reply:
Wouldn't that result in a default value of false?

Reply:
ok then how you have it isfine